versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/AbstractDatabaseAdapter.java

```diff
@@ -861,6 +876,18 @@ protected void checkForModifiedKeysBetweenExpectedAndCurrentCommit(
   protected abstract void writeIndividualCommit(OP_CONTEXT ctx, CommitLogEntry entry)
       throws ReferenceConflictException;
 
+  /**
+   * Updates the commit log entry, against the commit log hash. All values of the given {@link
+   * CommitLogEntry} can be considered valid and consistent. Since, it is an override operation, the
+   * last update will be final version of the entry.
+   *
+   * @param ctx
+   * @param entry
+   * @throws ReferenceNotFoundException
+   */
+  protected abstract void overrideCommitEntry(OP_CONTEXT ctx, CommitLogEntry entry)
```

Comment:
Definitely do not overwrite/update/modify any existing commit log entry. This would change the *original* commit -log-entry, right?
Merge + transplant write _new_ commit-log-entries.

Reply:
This is around the concern I raised earlier - 

> @snazy @nastra Changing the commit metadata is causing a deviation in the hash calculation here - AbstractDatabaseAdapter.java#L581. This deviation then in-turn leads to incorrect calculation of parents while making a merge.
> There are two approaches I could think of -
> a. Hash calculation to ignore updatable properties.
> b. Track referred hash, and use that for matching purposes.
> However, both of these solutions don't appear clean to me. Please suggest if you can think of something else.

I believe I misunderstood your [comment](https://github.com/projectnessie/nessie/pull/2146#issuecomment-927941173) properly, assuming suggestion was to change and persist - 
> Sure, you change the contents of the commits.
> You probably have to adopt the call-sites of copyCommits as well, so the updated commit-entries are persisted (your current PR does not). 


Let me take another attempt to explain the problem, if we take following example - 
![image](https://user-images.githubusercontent.com/57615655/135412930-ca04edb2-2fe9-484d-924b-690a21c27565.png)
Before this change, if there's a second merge from a different branch, only commits that have to be merged are the ones after first merge and the merge hash. The hash is used to do this match. Now that metadata is different, the hash at merge commit (in the main branch) is also different and the matching algorithm doesn't know the point of last merge and as a result, now we try to pull in all the commits from the incoming branch (which is incorrect, guardrail catches the conflict).

Out of the approaches, that I initially thought of - 

- a. Hash calculation to ignore updatable properties -> since metadata is generic in the adapter classes, we should rule out selective serialisation.
- b. Track referred hash, and use that for matching purposes -> May be add a "mergeHash" in CommitEntry, and enhance the matching logic to check if the incoming hash is equal to either mergeHash or commit hash.

Please let me know your thoughts. If you are available for a short 10 mins call today discussing this problem, it would be helpful.

Reply:
copyCommits does exactly what the method name suggests: it **copies** commits - means: it creates new commit-log-entries. Why update if those are new??

Again: we will **never** update commit-log-entries - those are immutable.

Reply:
I ran into an issue, which happens when there are multiple merges from a branch. Let me come up with a solution.